src/DynamoCore/Library/XmlDocumentationExtensions.cs

```diff
@@ -101,34 +103,55 @@ private static string CleanUpDocString(this string text)
-            string clrMemberName = GetMemberElementName(function);
+            var assemblyName = function.Assembly;
+            var fullyQualifiedName = GetMemberElementName(function);
+            if (!documentNodes.ContainsKey(fullyQualifiedName))
+                LoadDataFromXml(xml);
+
+            // Case for operators.
+            if (assemblyName == null)
+                return String.Empty;
 
-            // match clr member name
-            var match = xml.XPathEvaluate(
-                String.Format("string(/doc/members/member[@name='{0}']/{1})", clrMemberName, suffix));
+            var keyForOverloaded = documentNodes.Keys.
+                        Where(key => key.Contains(function.ClassName + "." + function.FunctionName)).FirstOrDefault();
 
-            if (match is String && !string.IsNullOrEmpty((string)match))
+            switch (property)
             {
-                return match as string;
-            }
+                case "summary":
+                    if (documentNodes.ContainsKey(fullyQualifiedName))
+                        return documentNodes[fullyQualifiedName].Summary;
 
-            // fallback, namespace qualified method name
-            var methodName = function.QualifiedName;
+                    // Fallback for overloaded methods.
+                    if (keyForOverloaded != null)
+                        return documentNodes[keyForOverloaded].Summary;
 
-            // match with fallback
-            match = xml.XPathEvaluate(
-                String.Format(
-                    "string(/doc/members/member[contains(@name,'{0}')]/{1})", methodName, suffix));
+                    return String.Empty;
 
-            if (match is String && !string.IsNullOrEmpty((string)match))
-            {
-                return match as string;
-            }
+                case "description":
+                    if (documentNodes.ContainsKey(fullyQualifiedName))
+                        if (documentNodes[fullyQualifiedName].Parameters.ContainsKey(paramName))
+                            return documentNodes[fullyQualifiedName].Parameters[paramName];
 
-            return String.Empty;
+                    // Fallback for overloaded methods.
+                    if (keyForOverloaded != null)
+                        if (documentNodes[keyForOverloaded].Parameters.ContainsKey(paramName))
+                            return documentNodes[keyForOverloaded].Parameters[paramName];
+
+                    return String.Empty;
+                case "search":
+                    if (documentNodes.ContainsKey(fullyQualifiedName))
+                        return documentNodes[fullyQualifiedName].SearchTags;
+
+                    // Fallback for overloaded methods.
+                    if (keyForOverloaded != null)
+                        return documentNodes[keyForOverloaded].SearchTags;
+
+                    return String.Empty;
+                default:
+                    return String.Empty;
+            }
```

Comment:
This method needs cleaning up, see the many calls to `documentNodes.ContainsKey(fullyQualifiedName)` and `documentNodes[keyForOverloaded]`? It should be made very straightforward:

```csharp
MemberDocumentNode documentNode = null;
if (documentNodes.ContainsKey(fullyQualifiedName))
    documentNode = documentNodes[fullyQualifiedName];
else
{
    var overloadedName = function.ClassName + "." + function.FunctionName;
    if (documentNodes.ContainsKey(overloadedName))
        documentNode = documentNodes[overloadedName];
}

if (documentNode == null)
    return string.Empty;

switch(property)
{
    case DocumentElementType.Summary:
        return documentNode.Summary;
    case DocumentElementType.Description:
        return documentNode.Parameters[...];

    // bla bla ...
    
    default: throw new ArgumentInvalidException("property");
}
```

Then instead of strings, we need to define the following `enum`:

```csharp
enum DocumentElementType
{
    Summary, Description, SearchTags
}
```